### PR TITLE
Allow custom names for the events

### PR DESCRIPTION
**Pull request by DZakh**

### CODE REVIEW

File: codegenerator/cli/src/config_parsing/system_config.rs [hunk 1]
```diff
@@ -660,18 +655,30 @@ impl Event {
         human_cfg_event: EventConfig,
         opt_abi: &Option<EvmAbi>,
     ) -> Result<Self> {
-        let event = Event::get_abi_event(&human_cfg_event.event, opt_abi)?.into();
+        let event: NormalizedEthAbiEvent =
+            Event::get_abi_event(&human_cfg_event.event, opt_abi)?.into();
 
         Ok(Event {
+            name: human_cfg_event.name.unwrap_or(event.0.name.to_owned()),
```
Comment (DZakh) [inline codegenerator/cli/src/config_parsing/system_config.rs:662]: I'll add validation in another PR together with the fix for another reported bug where codegen fails with contract name containing dashes.

**Review by JonoPrest — COMMENTED** [review]
I'm halfway through but will finish reviewing tomorrow morning 👍🏼

File: codegenerator/cli/src/hbs_templating/codegen_templates.rs [hunk 2]
```diff
@@ -358,14 +357,9 @@ pub struct EventTemplate {
 
 impl EventTemplate {
     pub fn from_config_event(config_event: &system_config::Event) -> Result<Self> {
-        let name = config_event
-            .get_event()
-            .name
-            .to_owned()
-            .to_capitalized_options();
+        let name = config_event.name.to_capitalized_options();
```
Comment (JonoPrest) [inline codegenerator/cli/src/hbs_templating/codegen_templates.rs:360]: I'm just trying to think if there is any place in the code where we need the orginal name of the event but I don't think so...

**Review by JonoPrest — APPROVED** [review]
Great implementation @DZakh, minimal changes, maximal effect 😄 🚀

**Pull request merged**